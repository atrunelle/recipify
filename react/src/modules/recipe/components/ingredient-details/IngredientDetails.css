.ingredient-details .ingredient-details__button {
  position: absolute;
  right: 60px;
  top: 0px; }
  .ingredient-details .ingredient-details__button:last-child {
    padding-right: 0; }
