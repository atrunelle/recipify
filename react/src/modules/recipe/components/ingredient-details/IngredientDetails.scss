.ingredient-details {  
  .ingredient-details__button {
    position: absolute;
    right: 60px;
    top: 0px;

    &:last-child {
      padding-right: 0;
    }
  }
}