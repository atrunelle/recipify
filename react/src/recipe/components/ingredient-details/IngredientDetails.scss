.ingredient-details {  
  .ingredient-details__button {
    opacity: 0;
    position: absolute;
    right: 16px;
    top: 0px;
    z-index: 2000;
  }

  &:hover {
    .ingredient-details__button {
      opacity: 1;
    }
  }
}