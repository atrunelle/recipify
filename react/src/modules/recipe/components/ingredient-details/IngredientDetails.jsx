import React, { Component } from 'react';
import { 
  ExpansionPanel,
  ExpansionPanelDetails,
  ExpansionPanelSummary,
  IconButton,
  Typography,
} from '@material-ui/core';
import DeleteIcon from '@material-ui/icons/Delete';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import IngredientNutrients from '../ingredient-nutrients/IngredientNutrients';
import './IngredientDetails.css';

class IngredientDetails extends Component {

  render() {
    return (
      <ExpansionPanel className="ingredient-details">
        <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
          <Typography
            component="h3">
            {this.props.ingredient.name} 
          </Typography>
          <Typography>
            { this.props.ingredient.nutrients.calories }cal for {this.props.ingredient.nutrients.totalWeight }gr
          </Typography>
          <IconButton
            color="secondary"
            className="ingredient-details__button"
            aria-label="Delete"
            id={`removeButton-${this.props.index}`}
            onClick={() => this.props.remove(this.props.index)}>
            <DeleteIcon />
          </IconButton>
        </ExpansionPanelSummary>
        <ExpansionPanelDetails>
          <IngredientNutrients nutrients={this.props.ingredient.nutrients}></IngredientNutrients>
        </ExpansionPanelDetails>
      </ExpansionPanel>
    );
  }
}

export default IngredientDetails;