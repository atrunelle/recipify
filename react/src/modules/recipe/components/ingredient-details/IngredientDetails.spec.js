import React from 'react';
import { shallow } from 'enzyme';
import IngredientDetails from './IngredientDetails';

describe('Component: IngredientDetails', () => {
  let component;
  const mockRemove = jest.fn();

  beforeEach(() => {
    const mockIngredient = {
      name: 'Carrot',
      nutrients: {
        calories: 100,
        totalWeight: 260,
        totalNutrients: {},
        totalDaily: {},
        dietLabels: [],
        healthLabels: [],
      }
    };

    component = shallow(
      <IngredientDetails 
        ingredient={mockIngredient}
        remove={mockRemove}
        index={1}/>
    );
  });

  it('should remove ingredient', () => {
    component.find('#removeButton-1').simulate('click');
    expect(mockRemove).toHaveBeenCalledWith(1);
  });

  it('should match snapshot', () => {
    expect(component.html()).toMatchSnapshot();
  });
});