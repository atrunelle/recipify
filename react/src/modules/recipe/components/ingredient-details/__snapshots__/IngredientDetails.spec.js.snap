// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: IngredientDetails should match snapshot 1`] = `"<div class=\\"MuiPaper-root-4 MuiPaper-elevation1-7 MuiExpansionPanel-root-1 ingredient-details\\"><div tabindex=\\"0\\" class=\\"MuiButtonBase-root-37 MuiExpansionPanelSummary-root-31\\" role=\\"button\\" aria-expanded=\\"false\\"><div class=\\"MuiExpansionPanelSummary-content-35\\"><h3 class=\\"MuiTypography-root-40 MuiTypography-body1-49\\">Carrot</h3><p class=\\"MuiTypography-root-40 MuiTypography-body1-49\\">100cal for 260gr</p><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-37 MuiIconButton-root-64 MuiIconButton-colorSecondary-67 ingredient-details__button\\" type=\\"button\\" aria-label=\\"Delete\\" id=\\"removeButton-1\\"><span class=\\"MuiIconButton-label-69\\"><svg class=\\"MuiSvgIcon-root-70\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><g><path d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z\\"></path></g></svg></span><span class=\\"MuiTouchRipple-root-76\\"></span></button></div><div tabindex=\\"-1\\" class=\\"MuiButtonBase-root-37 MuiIconButton-root-64 MuiExpansionPanelSummary-expandIcon-36\\" role=\\"button\\" aria-hidden=\\"true\\"><span class=\\"MuiIconButton-label-69\\"><svg class=\\"MuiSvgIcon-root-70\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><g><path d=\\"M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z\\"></path></g></svg></span><span class=\\"MuiTouchRipple-root-76\\"></span></div></div><div class=\\"MuiCollapse-container-83\\" style=\\"min-height:0px\\" aria-hidden=\\"true\\"><div class=\\"MuiCollapse-wrapper-85\\"><div class=\\"MuiCollapse-wrapperInner-86\\"><div class=\\"MuiExpansionPanelDetails-root-87\\"><div class=\\"MuiGrid-container-88 MuiGrid-spacing-xs-8-109 MuiGrid-align-items-xs-center-96 MuiGrid-justify-xs-center-105\\"><div class=\\"MuiGrid-container-88 MuiGrid-justify-xs-space-between-107\\"></div><div class=\\"MuiGrid-container-88 MuiGrid-align-items-xs-center-96 MuiGrid-justify-xs-center-105\\"></div></div></div></div></div></div></div>"`;
